src/vs/editor/contrib/wordHighlighter/browser/wordHighlighter.ts

```diff
@@ -168,6 +246,13 @@ function computeOccurencesAtPosition(registry: LanguageFeatureRegistry<DocumentH
 	return new TextualOccurenceAtPositionRequest(model, selection, wordSeparators);
 }
 
+function computeOccurencesMultiModel(registry: LanguageFeatureRegistry<MultiDocumentHighlightProvider>, model: ITextModel, selection: Selection, wordSeparators: string, otherModels: ITextModel[]): IOccurenceAtPositionRequest {
+	if (registry.has(model)) {
+		return new MultiModelOccurenceRequest(model, selection, wordSeparators, registry, otherModels);
+	}
+	return new TextualMultiModelOccurenceRequest(model, selection, wordSeparators, otherModels);
```

Comment:
Assuming you have an editor feature which implements the multi document occurences provider generically (my previous comment) then you don't need this nor the TextualMultiModel...Request because you can assume/guarantee that there is already at least one provider